### PR TITLE
fix: event filtering for empty or non-string event names 

### PR DESCRIPTION
**Pull request by MoumitaM**
## PR Description

Added a fix for the below use case:
When allowlist is selected and track events are triggered without the event name or the name is not a string we shouldn't let the event pass through that destination.

## Notion ticket

Ticket link

## Screenshots

Please add screenshots for any new features or UI bug fixes for the following browsers -

- Chrome
  >
- Firefox
  >
- Safari
  >

## Security

- [ ] The code changed/added as part of this pull request won't create any security issues with how the software is being used.

### CODE REVIEW

**Comment by bardisg** [comment]
@MoumitaM is this something we need to fix in v1.1 too?

File: packages/analytics-js-plugins/src/nativeDestinationQueue/utilities.ts [hunk 1]
```diff
@@ -13,29 +13,34 @@ import { DEFAULT_QUEUE_OPTIONS } from './constants';
...
   }
 
   const { blacklistedEvents, whitelistedEvents, eventFilteringOption } = dest.config;
-  // TODO: might have to make this logic case-sensitive
-  const formattedEventName = eventName.trim().toUpperCase();
+
   switch (eventFilteringOption) {
     // Blacklist is chosen for filtering events
     case 'blacklistedEvents':
+      if (!eventName || typeof eventName !== 'string') {
```
Comment (bardisg) [inline packages/analytics-js-plugins/src/nativeDestinationQueue/utilities.ts:26]: can the user blacklist all track events that do not have a name?

**Comment by MoumitaM** [comment]
> @MoumitaM is this something we need to fix in v1.1 too?

Yes, But this will be a breaking change

File: packages/analytics-js-plugins/src/nativeDestinationQueue/utilities.ts [hunk 2]
```diff
@@ -13,29 +13,34 @@ import { DEFAULT_QUEUE_OPTIONS } from './constants';
...
+        return false;
+      }
       if (Array.isArray(blacklistedEvents)) {
         return blacklistedEvents.some(
-          eventObj => eventObj.eventName.trim().toUpperCase() === formattedEventName,
+          eventObj => eventObj.eventName.trim().toUpperCase() === eventName.trim().toUpperCase(), // TODO: might have to make this logic case-sensitive
```
Comment (bardisg) [inline packages/analytics-js-plugins/src/nativeDestinationQueue/utilities.ts:31]: lets confirm with Ryan if we should make it case-sensitive

File: packages/analytics-js-plugins/src/nativeDestinationQueue/utilities.ts [hunk 3]
```diff
@@ -13,29 +13,34 @@ import { DEFAULT_QUEUE_OPTIONS } from './constants';
 const getNormalizedQueueOptions = (queueOpts: DestinationsQueueOpts): DestinationsQueueOpts =>
   mergeDeepRight(DEFAULT_QUEUE_OPTIONS, queueOpts);
 
-const isEventDenyListed = (eventName: Nullable<string>, dest: Destination) => {
-  if (!eventName || typeof eventName !== 'string') {
+const isEventDenyListed = (eventType: string, eventName: Nullable<string>, dest: Destination) => {
```
Comment (bardisg) [inline packages/analytics-js-plugins/src/nativeDestinationQueue/utilities.ts:16]: lets add unit tests for these new cases

File: packages/analytics-js-plugins/src/nativeDestinationQueue/utilities.ts (hunk 1 above)
Comment (MoumitaM) [reply packages/analytics-js-plugins/src/nativeDestinationQueue/utilities.ts:26]: No

File: packages/analytics-js-plugins/src/nativeDestinationQueue/utilities.ts (hunk 2 above)
Comment (MoumitaM) [reply packages/analytics-js-plugins/src/nativeDestinationQueue/utilities.ts:31]: Already asked, let's wait for his reply. Right now it is case in-sensitive.

File: packages/analytics-js-plugins/src/nativeDestinationQueue/utilities.ts (hunk 3 above)
Comment (saikumarrs) [inline packages/analytics-js-plugins/src/nativeDestinationQueue/utilities.ts:16]: `eventType` should be of `RudderEventType` type.

File: packages/analytics-js-plugins/src/nativeDestinationQueue/utilities.ts (hunk 1 above)
Comment (saikumarrs) [inline packages/analytics-js-plugins/src/nativeDestinationQueue/utilities.ts:26]: Let's extract this check into a method.

File: packages/analytics-js-plugins/src/nativeDestinationQueue/utilities.ts (hunk 3 above)
Comment (MoumitaM) [reply packages/analytics-js-plugins/src/nativeDestinationQueue/utilities.ts]: Addressed

Comment (MoumitaM) [reply packages/analytics-js-plugins/src/nativeDestinationQueue/utilities.ts]: Addressed

File: packages/analytics-js-plugins/src/nativeDestinationQueue/utilities.ts (hunk 1 above)
Comment (MoumitaM) [reply packages/analytics-js-plugins/src/nativeDestinationQueue/utilities.ts]: Addressed

**Review by saikumarrs — APPROVED** [review]

**Review by saikumarrs — APPROVED** [review]

**Review by bardisg — APPROVED** [review]

**Pull request merged**